src/Trakx.YouSign.ApiClient/ClientConfigurator.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Trakx.YouSign.ApiClient.Utils;
+using Trakx.Utils.Apis;
+
+namespace Trakx.YouSign.ApiClient
+{
+    internal class ClientConfigurator
+    {
+        private readonly IServiceProvider _provider;
+
+        public ClientConfigurator(IServiceProvider provider)
```

Comment:
Here I don't think this pattern isso nice actually. In the exchange client, it was done this way to allow getting a NoCredentialProvider for the Public client and a ApiKeyCredentialProvider for other clients.
I think you should be able to simply inject a ICredentialProvider and a YouSignApiConfiguration if you register the correct ones in the startup menu.
It is not a big deal but injecting the container is a sort of anti pattern.

Reply:
ClientConfigurator class was fully redesigned.